src/youtube.js

```diff
@@ -0,0 +1,1092 @@
+  //   this._eventManager.listen(this._mediaSourceAdapter, Html5EventType.WAITING, (event: FakeEvent) => this.dispatchEvent(event));
+  //   this._eventManager.listen(this._mediaSourceAdapter, CustomEventType.MEDIA_RECOVERED, (event: FakeEvent) => this.dispatchEvent(event));
+  // }
+  // }
+
+  /**
+   * Handles errors from the video element
+   * @returns {void}
+   * @private
+   */
+  _handleVideoError(): void {
+    this.dispatchEvent(new FakeEvent(EventType.ERROR, {}));
+  }
+
+  /**
+   * Remove the listeners of the video element events.
+   * @public
+   * @returns {void}
+   */
+  detach(): void {
+    Object.keys(EventType).forEach(html5Event => {
+      this._eventManager.unlisten(this._el, EventType[html5Event]);
+    });
+    // if (this._mediaSourceAdapter) {
+    //   this._eventManager.unlisten(this._mediaSourceAdapter, CustomEventType.VIDEO_TRACK_CHANGED);
+    //   this._eventManager.unlisten(this._mediaSourceAdapter, CustomEventType.AUDIO_TRACK_CHANGED);
+    //   this._eventManager.unlisten(this._mediaSourceAdapter, CustomEventType.TEXT_TRACK_CHANGED);
+    //   this._eventManager.unlisten(this._mediaSourceAdapter, CustomEventType.TEXT_CUE_CHANGED);
+    // }
+  }
+
+  /**
+   * @returns {HTMLVideoElement} - The video element.
+   * @public
+   */
+  getVideoElement(): HTMLVideoElement {
+    return this._el;
+  }
+
+  /**
+   * Select a new video track.
+   * @param {VideoTrack} videoTrack - The video track object to set.
+   * @returns {void}
+   */
+  selectVideoTrack(videoTrack: VideoTrack): void {
+    if (this._api) {
+
+      if (videoTrack instanceof VideoTrack && (!videoTrack.active || this.isAdaptiveBitrateEnabled()) ) {
+        if (this.isAdaptiveBitrateEnabled()) {
+          this.dispatchEvent(EventType.ABR_MODE_CHANGED, {mode: 'manual'});
+        }
+        this._isAdaptiveBitrate = false;
+        this._api.setPlaybackQuality(videoTrack.label);
+      }
+    }
+  }
+
+  /**
+   * Select a new audio track.
+   * @param {AudioTrack} audioTrack - The video track object to set.
+   * @returns {void}
+   */
+  selectAudioTrack(audioTrack: AudioTrack): void {
+    Youtube._logger.info(`audio track switching not supported, cannot switch to ` + audioTrack.language);
+  }
+
+  /**
+   * Select a new text track.
+   * @param {PKTextTrack} textTrack - The playkit text track object to set.
+   * @returns {void}
+   */
+  selectTextTrack(textTrack: PKTextTrack): void {
+    Youtube._logger.info(`text track switching not supported, cannot switch to ` + textTrack.language);
+  }
+
+  /**
+   * Hide the text track
+   * @function hideTextTrack
+   * @returns {void}
+   * @public
+   */
+  hideTextTrack(): void {}
+
+  /**
+   * Enables adaptive bitrate switching according to the media source extension logic.
+   * @function enableAdaptiveBitrate
+   * @returns {void}
+   * @public
+   */
+  enableAdaptiveBitrate(): void {
+    if (this._api) {
+      this._api.setPlaybackQuality('default');
```

Comment:
string to enum?

Reply:
string, they don't have enums.
In any case, as we have noticed, the API is just a stub and doesn't work.